File: flutter/lib/domain/path_finding/distances.dart
import 'dart:math';
import '../space/coord_2d.dart';

double calculateEuclideanDistance(Coord2D start, Coord2D end) {
  int dx = start.x - end.x;
  int dy = start.y - end.y;
  return sqrt((dx * dx + dy * dy).abs());
}

double calculateManhattanDistance(Coord2D start, Coord2D end) {
  return (start.x - end.x).abs() + (start.y - end.y).abs().toDouble();
}

File: jvm/ship-conquest/src/main/kotlin/com/example/shipconquest/domain/path_finding/distances.kt
package com.example.shipconquest.domain.path_finding

import com.example.shipconquest.domain.Vector2
import kotlin.math.abs
import kotlin.math.sqrt

fun calculateEuclideanDistance(start: Vector2, end: Vector2): Double {
    val dx = start.x - end.x
    val dy = start.y - end.y
    return sqrt(abs(dx * dx + dy * dy).toDouble())
}

fun calculateManhattanDistance(start: Vector2, end: Vector2): Double {
    return abs(start.x - end.x) + abs(start.y - end.y).toDouble()
}